scripts/conversion_script.py

```diff
@@ -36,6 +36,46 @@ def __init__(self, path="", host="", keep=False):
         self.path = path
         self.host = host
         self.keep = keep
+        self.backup_suffix = ".backup"
+
+    def create(self, data):
+        response = urlopen(self.host)
+        data = response.read()
+        directory = os.path.dirname(self.path)
+        if not os.path.exists(directory):
+            print("Creating directory at '%s'" % directory)
+            os.makedirs(directory, mode=0o755)
+
+        print("Writing file to destination: '%s'" % self.path)
+        with open(self.path, mode="w") as handler:
+            handler.write(data)
+            os.chmod(self.path, 0o644)
+
+    def delete(self):
+        if os.path.exists(self.path):
+            print("Removing the file '%s' as it was previously downloaded." % self.path)
+            os.remove(self.path)
+            return True
+        return False
+
+    def restore(self):
+        """Restores file backup (rename). Returns True if restored, otherwise False."""
+        if os.path.exists(self.path + self.backup_suffix):
+            print("Restoring backed up file %s." % (self.path))
+            os.rename(self.path + self.backup_suffix, self.path)
+            return True
+        return False
+
+    def backup(self):
+        """Creates backup file (rename). Returns True if backed up, otherwise False."""
+        if os.path.exists(self.path):
+            print(
+                "File %s already present on system, backing up to %s."
+                % (self.path, self.path + self.backup_suffix)
+            )
+            os.rename(self.path, self.path + ".backup")
+            return True
+        return False
```

Comment:
Also, can we change all the `if os.path.exists()` to `try...except:`? This can avoid race conditions of file creation/deletion and etc... If that fails because of the race condition, we should simply log it with `print()` and return the value we would expect.

Reply:
@andywaltlova

Reply:
yep .. have it on my TODO :) will take a look today

Reply:
@r0x0d please take a look :)

Reply:
looking at it right now! Thanks for the ping!